### PR TITLE
Switch to Pygit2

### PR DESCRIPTION
**Pull request by sushantmimani**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [x] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [x] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [x] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [ ] Yes (backward compatible)
* [x] No (breaking changes)
In its current version, this PR will flag findings for renamed files even if the findings in the new file(after rename) have been excluded from Tartufo.

## What's new?
This is the first PR in a series of PRs aimed at re-implementing #122. Current status:
- scan-remote-repo runs successfully
- scan-local-repo runs successfully
- pre-commit runs successfully
- All unit tests pass
- `--max-depth` option is unimplemented
- file renames are not tracked so extra findings ( which might have exclusions for the new file path) are generated

### CODE REVIEW

File: tartufo/scanner.py [hunk 1]
```diff
@@ -648,33 +620,35 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
...
+        except pygit2.GitError as exc:
+            raise types.GitRemoteException(str(exc)) from exc
 
         self.logger.debug(
             "Branches to be scanned: %s",
             ", ".join([str(branch) for branch in branches]),
         )
 
-        for branch in branches:
-            self.logger.info("Scanning branch: %s", branch)
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py:700]: Did this get dropped intentionally, or should it be reinserted between L633 and L634?

**Review by rbailey-godaddy — COMMENTED** [review]
Looking pretty good, although I bet it was a lot harder to come up with the changes than it was to read them. I noted one point where a message was dropped in the transformation -- I'm guessing that is unintentional because the look-and-feel appears unchanged everywhere else.

File: tartufo/scanner.py (hunk 1 above)
Comment (sushantmimani) [reply tartufo/scanner.py:700]: I dropped it accidentally. Added back now.

**Review by emayuri-godaddy — APPROVED** [review]
LGTM!

File: tartufo/scanner.py [hunk 2]
```diff
@@ -648,33 +625,36 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
...
-            for curr_commit, prev_commit in self._iter_branch_commits(
-                self._repo, branch
+            curr_commit: pygit2.Commit = None
+            prev_commit: pygit2.Commit = None
+            for curr_commit in self._repo.walk(
+                branch.resolve().target, pygit2.GIT_SORT_TOPOLOGICAL
             ):
-                diff_index = curr_commit.diff(prev_commit, create_patch=True)
+                if not curr_commit.parents:
+                    continue
```
Comment (tarkatronic) [inline tartufo/scanner.py]: What is this here for? What would a commit with no parents indicate?

Also we should probably add a debug log here.

**Review by rbailey-godaddy — APPROVED** [review]
LGTM. I didn't see anything weird at this point, and I trust the unit tests more than my deductive abilities in this area. ;-)

**Review by tarkatronic — CHANGES_REQUESTED** [review]
Just one small thing here and I think we're good for an MVP on this change.

**Review by tarkatronic — APPROVED** [review]
This will be a great starting point for a fast new tartufo! Thanks for all the work @jgowdy, @sushantmimani, and @emayuri-godaddy!

**Pull request merged**